packages/open-api-gateway/test/project/open-api-gateway-ts-project/regenerate-clients.test.ts

```diff
@@ -0,0 +1,336 @@
+
+    const generatedFileStatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+    const generatedDocHTML2StatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        "documentation",
+        DocumentationFormat.HTML2,
+        "index.html"
+      )
+    );
+
+    // current `index.ts` should have later modified timestamp as the original project's
+    expect(generatedFileStats.mtime.getTime()).toBeLessThan(
+      generatedFileStatsOther.mtime.getTime()
+    );
+    // current `index.html` in "html2" doc folder should be untouched
+    expect(generatedDocHTML2Stats.mtime).toEqual(
+      generatedDocHTML2StatsOther.mtime
+    );
+  });
+
+  it("Change spec will regenerate clients/docs", () => {
+    const project = new OpenApiGatewayTsProject({
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+      documentationFormats: [DocumentationFormat.HTML2],
+    });
+    project.synth();
+
+    const { generatedCodeDir, outdir, specDir, specFileName } = project;
+    const generatedFileStats = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+    const generatedDocHTML2Stats = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        "documentation",
+        DocumentationFormat.HTML2,
+        "index.html"
+      )
+    );
+
+    // change spec:
+    const genSpecFilePath = path.join(outdir, "src", specDir, specFileName);
+    const genSpecFileContent = fs.readFileSync(genSpecFilePath, {
+      encoding: "utf-8",
+    });
+    fs.writeFileSync(
+      genSpecFilePath,
+      genSpecFileContent.replace("/hello:", "/hello-changed:"),
+      { encoding: "utf-8" }
+    );
+
+    const projectOther = new OpenApiGatewayTsProject({
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+      outdir,
+    });
+    projectOther.synth();
+
+    const generatedFileStatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+    const generatedDocHTML2StatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        "documentation",
+        DocumentationFormat.HTML2,
+        "index.html"
+      )
+    );
+
+    // current `index.ts` should have later modified timestamp as the original project's
+    expect(generatedFileStats.mtime.getTime()).toBeLessThan(
+      generatedFileStatsOther.mtime.getTime()
+    );
+    // current `index.html` in "html2" doc folder should be untouched
+    expect(generatedDocHTML2Stats.mtime).toEqual(
+      generatedDocHTML2StatsOther.mtime
+    );
```

Comment:
Not for this pr but we should add some cleanup logic - the second project doesn't include any `documentationFormats` so ideally we would delete it :)

Reply:
agreed